backend/src/main/java/com/example/parkingspot/controller/PersonController.java

```diff
@@ -46,10 +46,10 @@ public ResponseEntity<Person> getPerson(@PathVariable(name = "id") String userId
   public ResponseEntity<Person> addNewPerson(@RequestBody Person person) {
 
     Person newUser = personService.registerNewPerson(person);
-    URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}").buildAndExpand(newUser.getId())
-        .toUri();
 
-    if (newUser != null) {
+    if (newUser.getId() > 0) {
+      URI location = ServletUriComponentsBuilder.fromCurrentRequest().path("/{id}").buildAndExpand(newUser.getId())
```

Comment:
What does this do?

Reply:
When registering a new Person, we dont specify an id. The id gets set på the database. This checks if the Person we are trying to create is returned from PersonService with an id greater then 0 (if its saved to a DB it always starts from 1 and count up, so 0 is just some easy check if create was successful).

If it has and id greater then 0, the endpoint/method will return a response with the created Person and a location URI.